src/aliased_struct.h

```diff
@@ -0,0 +1,73 @@
+#ifndef SRC_ALIASED_STRUCT_H_
+#define SRC_ALIASED_STRUCT_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_internals.h"
+#include "v8.h"
+#include <memory>
+
+namespace node {
+
+template <typename T,
+          typename = std::enable_if_t<std::is_object<T>::value>>
+class AliasedStruct {
+ public:
+  explicit AliasedStruct(v8::Isolate* isolate) : isolate_(isolate) {
+    const v8::HandleScope handle_scope(isolate);
+
+    store_ = v8::ArrayBuffer::NewBackingStore(isolate, sizeof(T));
+    ptr_ = new (store_->Data()) T;
```

Comment:
```suggestion
  template <typename... Args>
  explicit AliasedStruct(v8::Isolate* isolate, Args&&... args) : isolate_(isolate) {
    const v8::HandleScope handle_scope(isolate);

    store_ = v8::ArrayBuffer::NewBackingStore(isolate, sizeof(T));
    ptr_ = new (store_->Data()) T(std::forward<Args>(args)...);
```

Similar to what we did in b32fa7bcc69aec0acb63792c799a22eeae7424c6 since this is intended to be a generic utility